src/clients/dotnet/TigerBeetle/NativeClient.cs

```diff
@@ -11,6 +11,7 @@ namespace TigerBeetle;
 internal sealed class NativeClient : IDisposable
 {
     private volatile IntPtr client;
+    private readonly object syncRoot = new object();
```

Comment:
>The use of the NativeClient object as the mutex means a library
user could accidently lock the NativeClient and interfere with
tb_client synchronization. Instead, NativeClient now holds its
own internal object to acquire locks on.

Great write up King! After reading it, immediately I realized that the `NativeClient` instance is already private and never exposed to the final user (both in Java and C#) :facepalm:

So, there is no need for such pattern. You may completely ignore what I said before about using a private object as a lock, using `this` is just fine!